src/librustc_codegen_llvm/back/write.rs

```diff
@@ -651,7 +651,7 @@ pub(crate) unsafe fn codegen(
             let thin = ThinBuffer::new(llmod);
             let data = thin.data();
 
-            if config.emit_bc || config.obj_is_bitcode {
+            if config.emit_bc || config.emit_obj == EmitObj::Bitcode {
```

Comment:
This function in general is pretty gnarly, it's practically begging though to be a `match` statement of some kind. Would it be possible to transform most of this function into a `match config.emit_obj` and then do everything necessary in the arms?

Reply:
I agree it's gnarly! And that's after I already improved things in #70289.

But I don't see how to use a `match` here. The function is doing a lot of stuff that's not covered by `emit_obj`, and also there is overlap between various cases, rather than a clean disjunction.

Reply:
After staring at the code some more, I managed to use a `match config.emit_obj` at the end of the function for a slight readability improvement.